sktree/tree/_unsup_criterion.pyx

```diff
@@ -103,26 +171,40 @@ cdef class TwoMeans(UnsupervisedCriterion):
         samples : array-like, dtype=SIZE_t
             A mask on the samples, showing which ones we want to use
         """
-        # need to implement
-        pass
+        self.X = X
+        self.sample_weight = sample_weight
+        self.weighted_n_samples = weighted_n_samples
+        self.sample_indices = sample_indices
```

Comment:
The base class defines `samples`, not `sample_indices`, so this will fail compilation

Reply:
Thank you! The whole base class `init` seems to be remnants of my initial copy-paste. I made a placeholder for `init` while child classes are fully inheriting cinit from the base class